File: School/cis4500/quick.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

void swap (int *a, int *b){
    int temp = *a;
    *a = *b;
    *b = temp;
}

int partition(int array[], int low, int high){
    int pivot = array[low];
    int left = low;
    int i;
    for (i = low + 1; i < high; i++){
        if (array[i] < pivot){
            left++;
            swap(&(array[i]),&(array[left]));
        }
    }
    swap(&(array[low]),&(array[left]));
    return left;
}

void quickSort(int array[], int low, int high){
    if ( low < high ){
        int pivot = partition(array,low,high);
        quickSort(array,low,pivot);
        quickSort(array,pivot+1,high);
    }
}

int main(void){
    clock_t begin, end;
    double timeElapsed;
    srand(time(NULL));
    int numNums;
    int i;
    scanf("%d",&numNums);
    int * A = malloc(sizeof(int)*numNums);
    for (i = 0; i < numNums; i++){
        A[i] = rand() % 10000000;
    }
    begin = clock();
    quickSort(A,0,numNums+1);
    end = clock();
    timeElapsed = (double)(end - begin) / CLOCKS_PER_SEC;
    printf("\n time: %lf\n",timeElapsed);
    return 0;
}

File: School/cis4500/quick.jl
function partition(array, low, high)
    pivot = array[low]
    left = low
    for i = low+1:high
        if array[i] < pivot
            left = left + 1
            array[i], array[left] = array[left], array[i]
        end
    end
    array[low], array[left] = array[left], array[low]
    return left;
end

function quickSort(array, low, high)
    if low < high
        pivot = partition(array,low,high)
        quickSort(array,low,pivot)
        quickSort(array,pivot+1,high)
    end
end

s = int(readline(STDIN))
x = cell(s)
for i = 1:s
    x[i] = rand(1:10000000)
end
@time quickSort(x,1,s)